src/client/address_book/storage.h

```diff
@@ -54,6 +54,22 @@ namespace client {
 /// @class AddressBookDefaults
 /// @brief Default string constants used throughout address book
 struct AddressBookDefaults {
+  /// @enum Subscription
+  /// @brief Subscription type for where to load/save subscription addresses
+  /// @notes Scoped to prevent namespace pollution (otherwise, purely stylistic)
+  enum struct Subscription
+  {
+    Default,
+    User,
+    Private,
+  };
+
+  /// @alias AddressMap
+  /// @brief Maps human-readable hostname to an identity hash and subscription type
+  /// @notes Alias makes refactoring and use simpler
+  using AddressMap =
+      std::map<std::string, std::pair<kovri::core::IdentHash, Subscription> >;
```

Comment:
I'm not a fan of pairing the hash with the subscription type but it does make sense that a hash will be unique to a subscription. You can replace the existing doxygen notes (or add a details entry) with something more descriptive.

Reply:
My intent is to allow for the address map to reflect a database entry, and more easily support [subscription feeds](https://geti2p.net/spec/proposals/112-addressbook-subscription-feed-commands) (if we decide to implement). Will add that to the `details` doxygen comment.